.github/CODEOWNERS

```diff
@@ -372,6 +372,8 @@ examples/developer_examples @elastic/appex-sharedux
 examples/discover_customization_examples @elastic/kibana-data-discovery
 x-pack/plugins/discover_enhanced @elastic/kibana-data-discovery
 src/plugins/discover @elastic/kibana-data-discovery
+src/plugins/discover/public/context_awareness/profile_providers/log_document_profile @elastic/kibana-data-discovery @elastic/obs-ux-logs-team
```

Comment:
Hey @weltenwort , I remember there was a discussion regarding the code ownership. 

Thinking about that doesn't it make sense to have `olly` profile providers in a separate package?

Reply:
To my knowledge the state of discussions was to keep everything in a unified directory structure for now so it's easily discoverable. @davismcphee have there been any further conversations on this topic?

Reply:
Okay..got it. I remember @davismcphee's comment regarding keeping registeration inside Discover's directory structure but was not sure about the complete profile customization code.  Sorry if i did not have full context.

The reason I ask this is I am also working on plan for security solution on similar lines and would be soon working on similar changes.

Reply:
I definitely have a preference for keeping the registrations inside Discover, but I'm a bit more flexible on where the provider code actually lives if solution teams have strong preferences here. The main thing is that Data Discovery is expected to be responsible for the overall experience in Discover and aware of when things change, so wherever the code lives, we should share ownership in `CODEOWNERS`.

The organization in this PR makes sense to me because we only need to update `CODEOWNERS` for a single folder, but I'm open to discuss it if Security has strong opinions otherwise.

Reply:
>I definitely have a preference for keeping the registrations inside Discover

Makes sense. I agree.

>but I'm a bit more flexible on where the provider code actually lives if solution teams have strong preferences here

I do not have any strong opinions as I am still trying to understand and weigh in options.

Regarding the provider code, I am thinking it should *not* be too coupled of with discover's code. Provider code will be dependent on both `solution specific utils`(for example, o11y is separating out its utils in `kbn-data-view-utils` ) and discover code as well. 

Imo, If code remains separate, we will take care to extract discover's code in `discover-utils` package, otherwise it is very easy for provider code be dependent on discover's core code and later when it grows big, it will be difficult to take it out.

May be you all have already talked about it but I am just thinking out loud here.